src/const_encoded_len.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains the [`ConstEncodedLen`] trait and compliance tests.
+
+use crate::MaxEncodedLen;
+use core::{
+	marker::PhantomData,
+	num::*,
+	ops::{Range, RangeInclusive},
+	time::Duration,
+};
+use impl_trait_for_tuples::impl_for_tuples;
+
+/// Types that have a constant encoded length. This implies [`MaxEncodedLen`].
+///
+/// No derive macros is provided; instead use an empty implementation like for a marker trait.
+pub trait ConstEncodedLen: MaxEncodedLen {
+	/// Encoded size of any possible `Self` instance.
+	fn const_encoded_len() -> usize {
```

Comment:
As this is a marker trait and you already have the `MaxEncodedLen` as super trait, this doesn't require the method.